backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5680260_sys_gh14651_add_C_Project_unq_idx_ExternalID_OrgID_ProjectType.sql

```diff
@@ -0,0 +1,48 @@
+;
+
+-- 2023-03-02T17:36:26.035Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,8757,541318,540727,0,TO_TIMESTAMP('2023-03-02 19:36:25','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',30,TO_TIMESTAMP('2023-03-02 19:36:25','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:36:30.391Z
+CREATE UNIQUE INDEX ExternalID_OrgID_ProjectType_NonNull ON C_Project (ExternalId,AD_Org_ID,C_ProjectType_ID) WHERE C_ProjectType_ID IS NOT NULL and isactive = 'Y'
+;
+
+-- 2023-03-02T17:37:12.620Z
+INSERT INTO AD_Index_Table (AD_Client_ID,AD_Index_Table_ID,AD_Org_ID,AD_Table_ID,Created,CreatedBy,EntityType,IsActive,IsUnique,Name,Processing,Updated,UpdatedBy,WhereClause) VALUES (0,540728,0,203,TO_TIMESTAMP('2023-03-02 19:37:12','YYYY-MM-DD HH24:MI:SS'),100,'D','Y','Y','ExternalID_OrgID_ProjectType_Nullable','N',TO_TIMESTAMP('2023-03-02 19:37:12','YYYY-MM-DD HH24:MI:SS'),100,'C_ProjectType_ID IS NULL and isactive = ''Y''')
+;
+
+-- 2023-03-02T17:37:12.622Z
+INSERT INTO AD_Index_Table_Trl (AD_Language,AD_Index_Table_ID, ErrorMsg, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy,IsActive) SELECT l.AD_Language, t.AD_Index_Table_ID, t.ErrorMsg, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy,'Y' FROM AD_Language l, AD_Index_Table t WHERE l.IsActive='Y'AND (l.IsSystemLanguage='Y') AND t.AD_Index_Table_ID=540728 AND NOT EXISTS (SELECT 1 FROM AD_Index_Table_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Index_Table_ID=t.AD_Index_Table_ID)
+;
+
+-- 2023-03-02T17:37:24.198Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,584192,541319,540728,0,TO_TIMESTAMP('2023-03-02 19:37:24','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',10,TO_TIMESTAMP('2023-03-02 19:37:24','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:37:29.947Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,1350,541320,540728,0,TO_TIMESTAMP('2023-03-02 19:37:29','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',20,TO_TIMESTAMP('2023-03-02 19:37:29','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:37:39.720Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,8757,541321,540728,0,TO_TIMESTAMP('2023-03-02 19:37:39','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',30,TO_TIMESTAMP('2023-03-02 19:37:39','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2023-03-02T17:37:43.370Z
+CREATE UNIQUE INDEX ExternalID_OrgID_ProjectType_Nullable ON C_Project (ExternalId,AD_Org_ID,C_ProjectType_ID) WHERE C_ProjectType_ID IS NULL and isactive = 'Y'
+;
```

Comment:
postgres sees `null` values as distinct, so the index must only on C_Project (ExternalId,AD_Org_ID)

Reply:
maybe also change the name to `ExternalID_OrgID_with_null_project_type`

Reply:
index renamed

Reply:
@fp-ps @cp-ps you can use `ExternalId,AD_Org_ID,coalesce(C_ProjectType_ID,0)` 
so maybe not two indixes are needed after all?